### PR TITLE
 Version bump updates

### PR DESCRIPTION
**Pull request by fmoral2**
#### Types of Changes ####
 
- debug mode added on template
- fixed 2 issues related to concurrency  
- Here we was unnecessaryly spinning up routines based on CMDS , which with a lot of commands and more than 1 ip, started to flood ssh and returning error for handshake 
* src/github.com/rancher/distros-test-framework/shared/aux.go:57 ERRO[2024-02-21T15:03:58Z] ssh: handshake failed: EOF Last call: github.com/rancher/distros-test-framework/shared.configureSSH in file:/go/src/github.com/rancher/distros-test-framework/shared/aux.go:182 ERRO[2024-02-21T15:03:58Z] ssh: handshake failed: EOF

https://github.com/rancher/distros-test-framework/blob/3311bc26a363a4418f3a05eeafebc032d2fb49e9/pkg/template/processor.go#L36 

 - fixed an issue here https://github.com/rancher/distros-test-framework/blob/3311bc26a363a4418f3a05eeafebc032d2fb49e9/pkg/template/processor.go#L67
which was sending a cmd that returns  the node info not parsed correctly , causing  to pass by this info to next function returning error related to not finding the host 

- fixed general error formatting wrapper  on the PKG 
- added a job that runs most of all general bumps tested
 
- fixed expectedValue result formatting 

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
yes - 

2. Verify code lint; we should not have errors.
yes - 

3. Update the documentation if needed.
yes- 

4. Update makefile and docker run if adding new tests.
yes 

5. Run your tests at least 4 times with all configurations needed and possible.
yes 

6. If needed test with different os types.

OS TESTED

Red Hat Enterprise Linux 8.9 (Ootpa)

Red Hat Enterprise Linux  9.2 

 Rocky Linux 8.9 (Green Obsidian)

 Oracle Linux Server 9.3 
ORACLE 8.9

SUSE Linux Enterprise Server 15 SP4

ubuntu

### CODE REVIEW

**Review by mdrahman-suse — DISMISSED** [review]
LGTM, one suggestion though, not sure if it will be bit overhead but I wanted to suggest to have the commands in a file so that they can be maintained separately and can be reused in other tests as needed, thoughts? 
For ex: commands in this file https://github.com/rancher/distros-test-framework/blob/bac6b325f52ae7b162c70ee7090745aa1b7f45f6/entrypoint/versionbump/componentsbump_test.go

**Comment by fmoral2** [comment]
> LGTM, one suggestion though, not sure if it will be bit overhead but I wanted to suggest to have the commands in a file so that they can be maintained separately and can be reused in other tests as needed, thoughts? For ex: commands in this file https://github.com/rancher/distros-test-framework/blob/bac6b325f52ae7b162c70ee7090745aa1b7f45f6/entrypoint/versionbump/componentsbump_test.go


Hey i have used in this way here since they are all used in the same file

But having a separated file for that we need to discuss a bit further , since normally in go we tend to use the consts and var very near to its usage, to increase readability and maintenance 

so depending for which commands are you saying , maybe the " cleaner " and maintainable way to use would be having a function that holds your used cmd options, and then u just call it and construct your command accordingly to each need ( in your consumer ) 
so in that way we are  "centralizing "  the cmd creation and maintenance and not creating a whole file for that 
https://github.com/rancher/qa-tasks/issues/967

https://github.com/rancher/distros-test-framework/blob/2ff8ae8ad52acf8d6fb3347f4b6bc57a59b1ada8/shared/cluster.go#L139C1-L139C5


BUT i also think that is not a bad idea , specially if the commands **are not exported**, which i dont think it will be the case , since we use cmds, through  out the whole project 

so looking at our use case and how things work on the framework x coding best practices on declaration , i would say the good approach would be using stuff like the function above on the file pasted 

BUT thats my POV, we should  definitely discuss that with team

File: shared/cluster.go
Comment (mdrahman-suse) [inline shared/cluster.go:244]: Did we update this method in the tests where this is being used?

Comment (mdrahman-suse) [reply shared/cluster.go:244]: Ok I see we did

**Review by mdrahman-suse — DISMISSED** [review]
LGTM

**Review by VestigeJ — DISMISSED** [review]

File: entrypoint/versionbump/componentsbump_test.go
Comment (ShylajaDevadiga) [inline entrypoint/versionbump/componentsbump_test.go:109]: Do we need sleep 20 here for sed?

File: entrypoint/versionbump/componentsbump_test.go
Comment (ShylajaDevadiga) [inline entrypoint/versionbump/componentsbump_test.go:118]: Could we add sleep with the restart cmd instead of having it as a separate cmd?

File: entrypoint/versionbump/versionbump_suite_test.go
Comment (ShylajaDevadiga) [inline entrypoint/versionbump/versionbump_suite_test.go:63]: Would prefer TestVersionBumpSuite instead

File: modules/install/rke2_master.sh
Comment (ShylajaDevadiga) [inline modules/install/rke2_master.sh:7]: Let's remove it if it is not needed.

File: entrypoint/versionbump/componentsbump_test.go
Comment (fmoral2) [reply entrypoint/versionbump/componentsbump_test.go:109]: probably we dont , but i will double check

File: entrypoint/versionbump/componentsbump_test.go
Comment (fmoral2) [reply entrypoint/versionbump/componentsbump_test.go:118]: this one i would like to keep separate just to make more clear, like if we have any error from restart we will know thats from there , rather than go to logs to find out if the "  nok " answer is from restart or from some other network/connection issue that might happen when trying to sleep 

but its just to separate concerns nothing else .

File: modules/install/rke2_master.sh
Comment (fmoral2) [reply modules/install/rke2_master.sh:7]: fixed tks

File: entrypoint/versionbump/componentsbump_test.go
Comment (fmoral2) [reply entrypoint/versionbump/componentsbump_test.go:109]: apparently i have added because after the sed, if you go try to restart right after it kind of stucks there
-- without sleep 
https://jenkins.int.rancher.io/job/rke2-tests/job/rke2-all-version-bump-validation/11/console

-with sleep 
https://jenkins.int.rancher.io/job/rke2-tests/job/rke2-all-version-bump-validation/12/console

Comment (fmoral2) [reply entrypoint/versionbump/componentsbump_test.go:109]: @ShylajaDevadiga

**Review by VestigeJ — APPROVED** [review]

**Review by mdrahman-suse — DISMISSED** [review]
LGTM, will wait for Shylaja's approval as she had few comments

File: entrypoint/versionbump/componentsbump_test.go
Comment (ShylajaDevadiga) [reply entrypoint/versionbump/componentsbump_test.go:109]: It seems to have failed on systemctl restart rke2-server and the test times out.

File: entrypoint/versionbump/componentsbump_test.go
Comment (ShylajaDevadiga) [inline entrypoint/versionbump/componentsbump_test.go:113]: Could we add sleep here instead?

File: modules/install/join_rke2_master.sh
Comment (rancher-max) [inline modules/install/join_rke2_master.sh:102]: Can we have the etcd version in a variable so that it's easier to update? (in all the places where we're doing this)

File: modules/install/join_rke2_master.sh
Comment (rancher-max) [inline modules/install/join_rke2_master.sh:145]: Do we need this to run on every install? This feels like it'll be good in the etcd version test, but outside of that, I don't think we need to do these validations during the install here as part of this script. Just having the sleep alone will slow things down.

File: modules/install/join_rke2_master.sh
Comment (rancher-max) [inline modules/install/join_rke2_master.sh:182]: We should only install etcdctl on etcd nodes. This will currently run on cp-only nodes as well if that is passed in the configuration.

File: modules/install/k3s_master.sh
Comment (rancher-max) [inline modules/install/k3s_master.sh:255]: For k3s, we should also only install etcdctl when using an etcd DB. So when externaldb is passed, we should not run this at all (in addition to the other comments made on one of the rke2 files).

File: pkg/assert/validate.go
Comment (rancher-max) [inline pkg/assert/validate.go:131]: Can we update the tests themselves so that they print their own results? This feels like an arbitrary call that is prone to miss needed changes when changes are made to those tests.

File: shared/aux.go
Comment (rancher-max) [inline shared/aux.go:489]: I think we have this already with `PrintClusterState`

**Review by rancher-max — COMMENTED** [review]
I think we should leave the CNI version bumps separate. It's not really valid to just change the config.yaml to a different CNI as they affect the host as well and can conflict with one another. Small ref: https://www.suse.com/support/kb/doc/?id=000020904.

File: modules/install/join_rke2_master.sh
Comment (fmoral2) [reply modules/install/join_rke2_master.sh:145]: i just added here thinking that we might find good to have etcdctl available where needed , now or future 

but if you think there is no reason, i can add that on the test itself. let me know  @rancher-max

File: pkg/assert/validate.go
Comment (fmoral2) [reply pkg/assert/validate.go:131]: tbh i dont want to add anything like that in any test, i want to have the flexibilty to only call where i want when i want , if i understood correctly your request.

so in this way, we only call this gather and print of results, in where we want.

File: shared/aux.go
Comment (fmoral2) [reply shared/aux.go:489]:
```
	cmd := "kubectl get nodes,pods -A -o wide --kubeconfig=" + KubeConfigFile
```
PrintClusterState is only print that, not all other  resources

File: modules/install/join_rke2_master.sh
Comment (fmoral2) [reply modules/install/join_rke2_master.sh]: done

File: modules/install/k3s_master.sh
Comment (fmoral2) [reply modules/install/k3s_master.sh]: done

File: modules/install/k3s_master.sh
Comment (mdrahman-suse) [inline modules/install/k3s_master.sh:120]: We can add the `install_etcdctl` here before checking channel

File: modules/install/join_rke2_master.sh
Comment (mdrahman-suse) [inline modules/install/join_rke2_master.sh:100]: Should remove this commented code?

File: modules/install/join_k3s_master.sh
Comment (mdrahman-suse) [inline modules/install/join_k3s_master.sh:97]: Should remove the commented code?

File: modules/install/join_rke2_master.sh
Comment (fmoral2) [reply modules/install/join_rke2_master.sh]: yes, tk you !

File: modules/install/k3s_master.sh
Comment (fmoral2) [reply modules/install/k3s_master.sh:120]: good call, fixed

File: docs/version_bump_template.md
Comment (mdrahman-suse) [inline docs/version_bump_template.md:80]: should be `v1.27...`?

File: entrypoint/versionbump/componentsbump_test.go
Comment (fmoral2) [reply entrypoint/versionbump/componentsbump_test.go:33]: added dns, etcd i removed everything as it was getting too much out of scope due to changes required

**Comment by fmoral2** [comment]
https://github.com/rancher/distros-test-framework/pull/74#discussion_r1556334143
@rancher-max  
i was reading your comment , looking at the function but thinking in another one. lol sorry 
yes totally, fixed.

File: entrypoint/versionbump/componentsbump_test.go
Comment (fmoral2) [reply entrypoint/versionbump/componentsbump_test.go:33]: ADDING top nodes ,pods as well

**Review by rancher-max — DISMISSED** [review]
LGTM

File: docs/version_bump_template.md
Comment (fmoral2) [reply docs/version_bump_template.md:80]: i had fixed somewhere else, doing now 
tks

**Review by rancher-max — APPROVED** [review]

**Review by mdrahman-suse — APPROVED** [review]
LGTM

**Pull request merged**
